### DISCUSSION TITLE
App creates public share

### DISCUSSION THREAD

**Discussion by ETCore7** [jy0205/Pyramid-Flow#215] [Ideas]
Thanks for Pyramid Flow, but the app creates a public share on *.gradio.live since FRP is invoked after creating the localhost. It shouldn't do that for local use. Also, the HF/FRP devs suggest to use your own controlled server if a public share is needed.
[github.com/huggingface/frp](https://github.com/huggingface/frp)
![GetTIkfW8AEsnD_](https://github.com/user-attachments/assets/135aca60-4f3d-42ca-a4cf-0758aeb0175e)
![GetUtYKWUAEBePA](https://github.com/user-attachments/assets/1c2c961e-3fa1-4fbf-8adc-755aa4c3cdc9)

**Comment by halr9000** [comment]
Definitely recommend not to enable sharing by default

**Comment by feifeiobama** [comment]
Thanks for your suggestion, we have enabled this as our cluster does not allow access with ssh port forwarding. Feel free to open a pull request and we will merge it as soon as possible.

**Comment by ETCore7** [comment]
Thank you. I don't know what that means, I am not a developer. PyramidFlow runs in Pinokio by '[Cocktail Peanut](https://x.com/cocktailpeanut/status/1867682712145850536)', who suggested to bring up the issue here.

**Reply by VyneNave** [reply]
Fixed it.  Gradio has an option to public share by adding the launch argument "share=True"

**Comment by ETCore7** [comment]
Thank you
